stats/opentelemetry/e2e_test.go

```diff
@@ -151,6 +118,12 @@ func (s) TestMethodTargetAttributeFilter(t *testing.T) {
 	}
 	rm := &metricdata.ResourceMetrics{}
 	reader.Collect(ctx, rm)
+	gotMetrics := map[string]metricdata.Metrics{}
+	for _, sm := range rm.ScopeMetrics {
+		for _, m := range sm.Metrics {
+			gotMetrics[m.Name] = m
+		}
+	}
 
 	wantMetrics := []metricdata.Metrics{
```

Comment:
Is there a reason to not use `testutils.MetricData` here?

Reply:
Yeah, the creation function specifies the full emitted metric atoms, and this one is a lighter weight check on one metric from client and one metric from server. Related to that, do you like my metrics data creation function? I added so many options to it (like our xDS Resources) and I had a ton of branches in the helper (and coupled some logic there with callsite like messages sent which need to be observed on peer). I don't really see a way around it, but let me know if you've seen a cleaner way to do this.

Reply:
Left some comments there :)
I feel like there is room for simplification. A good start though.